### PR TITLE
CSV Feed API module + Bambenek Consulting

### PR DESCRIPTION
**Pull request by liorblob**
## Status
Ready

## Related Issues
https://github.com/demisto/etc/issues/21019

## Required version of Demisto
5.5

## Must have
- [X] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [Bambenek Consulting YAML file](https://github.com/demisto/content/compare/csv-module?expand=1#diff-0e8dd2a988aeea04921e0274170a2251)

### CODE REVIEW

**Comment by kirbles19** [comment]
@liorblob Done

File: Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    feed_name = args.get('feed_name', 'CSV')
+    return hr, {f'{feed_name}.Indicator': entry_result}, indicators_list
+
+
+def feed_main(feed_name):
+    params = {k: v for k, v in demisto.params().items() if v is not None}
+    handle_proxy()
+    client = Client(**params)
+    command = demisto.command()
+    if command != 'fetch-indicators':
```
Comment (ronykoz) [inline Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py]: add a comment for why we used this condition

File: Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            args['feed_name'] = feed_name
+            readable_output, outputs, raw_response = commands[command](client, args)
+            return_outputs(readable_output, outputs, raw_response)
+    except Exception as e:
+        err_msg = f'Error in {feed_name} Integration - Encountered an issue with createIndicators' if \
+            'failed to create' in str(e) else f'Error in {feed_name} Integration [{e}]'
+        if command == 'fetch-indicators':
+            raise Exception(err_msg)
+        else:
+            return_error(err_msg)
```
Comment (ronykoz) [inline Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py]: why not to update the return_error?

File: Packs/ApiModules/Scripts/CSVFeedApiModule/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+## Common logic for CSV Feed integrations
+
+### Contents
+*  Client - client to build iterators from the CSV endpoints
```
Comment (ronykoz) [inline Packs/ApiModules/Scripts/CSVFeedApiModule/README.md]:
```suggestion
* Client - client to build iterators from the CSV endpoints
```

File: Packs/BambenekConsultingFeed/Integrations/BambenekConsultingFeed/BambenekConsultingFeed.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+def main():
+    # Main execution of the CSV API Module.
+    # This function allows to add to or override this execution.
+    feed_main('BambenekConsultingFeed')
```
Comment (ronykoz) [inline Packs/BambenekConsultingFeed/Integrations/BambenekConsultingFeed/BambenekConsultingFeed.py]: why not separated words?

File: Packs/BambenekConsultingFeed/Integrations/BambenekConsultingFeed/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+## [Unreleased]
+
+
+## [19.11.1] - 2019-11-26
+These notes will be published in the next release notes
```
Comment (ronykoz) [inline Packs/BambenekConsultingFeed/Integrations/BambenekConsultingFeed/CHANGELOG.md]: remove

File: Packs/BambenekConsultingFeed/Integrations/BambenekConsultingFeed/BambenekConsultingFeed.py (hunk 4 above)
Comment (liorblob) [reply Packs/BambenekConsultingFeed/Integrations/BambenekConsultingFeed/BambenekConsultingFeed.py]: for the context output

File: Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py (hunk 2 above)
Comment (liorblob) [reply Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py]: in a different PR

File: Packs/BambenekConsultingFeed/Integrations/BambenekConsultingFeed/BambenekConsultingFeed.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+def main():
+    # Main execution of the CSV API Module.
+    # This function allows to add to or override this execution.
+    params = {k: v for k, v in demisto.params().items() if v is not None}
```
Comment (ronykoz) [inline Packs/BambenekConsultingFeed/Integrations/BambenekConsultingFeed/BambenekConsultingFeed.py]: why is it done here and not in the generic one

File: Packs/FeedCSV/README.md [hunk 7]
```diff
@@ -0,0 +1 @@
+Pack for the generic CSV feed integration.
```
Comment (ronykoz) [inline Packs/FeedCSV/README.md]: change it to be more detailed please

File: Packs/FeedBambenekConsulting/Integrations/FeedBambenekConsulting/README.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
+## Overview
```
Comment (ronykoz) [inline Packs/FeedBambenekConsulting/Integrations/FeedBambenekConsulting/README.md]: we might want to add the sub feeds details here, like you have in the mimecast-node

**Comment by ronykoz** [comment]
@liorblob when you add the descriptions in the md files please go over it with Alex

**Comment by liorblob** [comment]
@kirbles19 In https://github.com/demisto/content/pull/5425/files#diff-cd44f8924c9a474d7e54b430f55a5140 we've added descriptions from minemeld. Worth going over?

File: Packs/ApiModules/Scripts/CSVFeedApiModule/test_data/ip_ranges.txt [hunk 9]
```diff
@@ -1,39 +1,39 @@
...
-1.1.1.1
-127.0.0.2
-22.22.22.22
-19.12.13.11
-192.168.10.1
-147.75.33.100
-95.181.53.78
-216.3.128.82
\ No newline at end of file
+7.7.7.7,a
```
Comment (ronykoz) [inline Packs/ApiModules/Scripts/CSVFeedApiModule/test_data/ip_ranges.txt]: why all the `,a` ?

File: Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py [hunk 10]
```diff
@@ -192,7 +217,7 @@ def get_indicators_command(client, args):
     limit = int(args.get('limit'))
     indicators_list = fetch_indicators_command(client, itype)
     entry_result = camelize(indicators_list[:limit])
-    hr = tableToMarkdown('Indicators', entry_result, headers=['Value', 'Type', 'Rawjson'])
+    hr = tableToMarkdown('Indicators', entry_result, headers=['value', 'type', 'rawJSON'])
```
Comment (ronykoz) [inline Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py]: does that come out ok when printing the rawJSON?

File: Packs/ApiModules/Scripts/CSVFeedApiModule/test_data/ip_ranges.txt (hunk 9 above)
Comment (liorblob) [reply Packs/ApiModules/Scripts/CSVFeedApiModule/test_data/ip_ranges.txt]: to test the mapping

File: Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py (hunk 10 above)
Comment (liorblob) [reply Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py]: what do you mean?

Comment (ronykoz) [reply Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py]: will it print the values properly if thats a large dictionary?

Comment (liorblob) [reply Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py]: hmm probably not. Should I remove it from the HR?

Comment (ronykoz) [reply Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py]: Yeah I think so, in the worst case we will add it back later

File: Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py [hunk 11]
```diff
@@ -164,26 +187,28 @@ def module_test_command(client: Client, args):
...
+                    if not indicator_type:
                         indicator_type = default_indicator_type
                     raw_json['type'] = indicator_type
-                    indicators.append({
-                        "value": value,
-                        "type": indicator_type,
-                        "rawJSON": raw_json,
-                    })
+
+                    indicator = {field: raw_json[key] for key, field in mapping.items()}
```
Comment (ronykoz) [inline Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py]: add this to custom fields

**Pull request merged**